### PR TITLE
Document Hoogle Plugin

### PR DESCRIPTION
**Pull request by fendor**
Adds only documentation.

### CODE REVIEW

File: src/Haskell/Ide/Engine/Plugin/Hoogle.hs [hunk 1]
```diff
@@ -114,12 +135,30 @@ renderTarget t = T.intercalate "\n\n" $
 
 ------------------------------------------------------------------------
 
+-- | Search for modules that satisfy the given search text.
+-- Will return at most five, unique results.
+--
+-- If an error happens, such as no has been hoogle database found,
```
Comment (lorenzo) [inline src/Haskell/Ide/Engine/Plugin/Hoogle.hs]: The grammar here is a bit off

File: src/Haskell/Ide/Engine/Plugin/Hoogle.hs [hunk 2]
```diff
@@ -114,12 +135,30 @@ renderTarget t = T.intercalate "\n\n" $
...
+-- an empty list will be returned.
 searchModules :: T.Text -> IdeM [T.Text]
 searchModules = fmap (nub . take 5) . searchTargets (fmap (T.pack . fst) . targetModule)
 
+-- | Search for packages that satisfy the given search text.
+-- Will return at most five, unique results.
+--
+-- If an error happens, such as no has been hoogle database found,
```
Comment (lorenzo) [inline src/Haskell/Ide/Engine/Plugin/Hoogle.hs]: Same here

File: src/Haskell/Ide/Engine/Plugin/Hoogle.hs [hunk 3]
```diff
@@ -114,12 +135,30 @@ renderTarget t = T.intercalate "\n\n" $
...
 searchPackages :: T.Text -> IdeM [T.Text]
 searchPackages = fmap (nub . take 5) . searchTargets (fmap (T.pack . fst) . targetPackage)
 
+-- | Search for Targets that fit to the given Text and satisfy the given predicate.
+-- Limits the amount of matches to at most ten. 
+-- Applies the predicate to the first ten matches. May also return zero matches, 
+-- although there are matches, if none of the first ten matches
+-- satisfies the predicate.
+--
+-- If an error happens, such as no hoogle database found,
```
Comment (lorenzo) [inline src/Haskell/Ide/Engine/Plugin/Hoogle.hs]: Same

File: src/Haskell/Ide/Engine/Plugin/Hoogle.hs (hunk 2 above)
Comment (fendor) [reply src/Haskell/Ide/Engine/Plugin/Hoogle.hs]: Thanks for catching that!

**Pull request merged**